src/pages/ReimbursementAccount/ACHContractStep.js

```diff
@@ -36,74 +38,106 @@ const propTypes = {
+                    errors[`beneficialOwner.${ownerID}.dob`] = this.props.translate('bankAccount.error.age');
+                }
+
+                if (!ValidationUtils.isRequiredFulfilled(values[`beneficialOwner.${ownerID}.ssnLast4`])
+                    || !ValidationUtils.isValidSSNLastFour(values[`beneficialOwner.${ownerID}.ssnLast4`])) {
+                    errors[`beneficialOwner.${ownerID}.ssnLast4`] = this.props.translate('bankAccount.error.ssnLast4');
+                }
+
+                if (!ValidationUtils.isRequiredFulfilled(values[`beneficialOwner.${ownerID}.street`])) {
+                    errors[`beneficialOwner.${ownerID}.street`] = this.props.translate('bankAccount.error.address');
+                }
+
+                if (values[`beneficialOwner.${ownerID}.street`]
+                    && !ValidationUtils.isValidAddress(values[`beneficialOwner.${ownerID}.street`])) {
+                    errors[`beneficialOwner.${ownerID}.street`] = this.props.translate('bankAccount.error.addressStreet');
+                }
+
+                if (!ValidationUtils.isRequiredFulfilled(values[`beneficialOwner.${ownerID}.city`])) {
+                    errors[`beneficialOwner.${ownerID}.city`] = this.props.translate('bankAccount.error.addressCity');
+                }
+
+                if (!ValidationUtils.isRequiredFulfilled(values[`beneficialOwner.${ownerID}.state`])) {
+                    errors[`beneficialOwner.${ownerID}.state`] = this.props.translate('bankAccount.error.addressState');
+                }
+
+                if (!ValidationUtils.isRequiredFulfilled(values[`beneficialOwner.${ownerID}.zipCode`])
+                    || !ValidationUtils.isValidZipCode(values[`beneficialOwner.${ownerID}.zipCode`])) {
+                    errors[`beneficialOwner.${ownerID}.zipCode`] = this.props.translate('bankAccount.error.zipCode');
+                }
+            });
         }
 
-        const errors = {};
-        _.each(this.requiredFields, (inputKey) => {
-            if (ValidationUtils.isRequiredFulfilled(this.state[inputKey])) {
-                return;
-            }
+        if (!ValidationUtils.isRequiredFulfilled(values.acceptTermsAndConditions)) {
+            errors.acceptTermsAndConditions = this.props.translate('common.error.acceptedTerms');
+        }
 
-            errors[inputKey] = true;
-        });
-        BankAccounts.setBankAccountFormValidationErrors({...errors, beneficialOwnersErrors});
-        return _.every(beneficialOwnersErrors, _.isEmpty) && _.isEmpty(errors);
+        if (!ValidationUtils.isRequiredFulfilled(values.certifyTrueInformation)) {
+            errors.certifyTrueInformation = this.props.translate('beneficialOwnersStep.error.certify');
+        }
+
+        return errors;
     }
 
-    removeBeneficialOwner(beneficialOwner) {
+    removeBeneficialOwner(ownerID) {
```

Comment:
Any method with an argument or return value should have a JSDoc. Could we please add one now?